frontend/packages/console-shared/src/hooks/promise-handler.ts

```diff
@@ -0,0 +1,17 @@
+import * as React from 'react';
+
+export const usePromiseHandler: PromisHandlerHook = () => {
+  const [inProgress, setInProgress] = React.useState(false);
+  const [errorMessage, setErrorMessage] = React.useState('');
+  const handlePromise = (promise) => {
+    setInProgress(true);
+    return promise
+      .then(() => setErrorMessage(''))
+      .catch((reason) => setErrorMessage(reason || 'An error occurred. Please try again.'))
```

Comment:
Even though we're handling the failed promise here, I think this will cause an unhandled promise rejection event (and an error overlay in dev mode) because we're calling `then` on the returned promise. We should test.

Reply:
@spadgett  This wasn't causing an uncaught Promise rejection, but it was "swallowing" the rejection since I wasn't returning a rejected promise or throwing. Should be fixed now.